src/Native/Unix/System.Security.Cryptography.Native/opensslshim.c

```diff
@@ -41,36 +56,45 @@ static bool OpenLibrary()
 
     if ((versionOverride != NULL) && strnlen(versionOverride, MaxVersionStringLength + 1) <= MaxVersionStringLength)
     {
+#ifdef __APPLE__
+        char soName[sizeof(DYLIBNAME_PREFIX) + MaxVersionStringLength + sizeof(DYLIBNAME_SUFFIX)] =
```

Comment:
Don't forget room for the null terminator just in case `strlen(versionOverride) == MaxVersionStringLength`.

Reply:
sizeof already takes it into account against a literal string.  This one is actually one bigger than it needs to be, because it has space for two null terminators.